<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerTransformsModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerTransformsModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>391</width>
    <height>1223</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Transforms</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../qSlicerTransformsModule.qrc">
    <normaloff>:/Icons/Transforms.png</normaloff>:/Icons/Transforms.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="TransformNodeSelectorLabel">
       <property name="text">
        <string>Active Transform:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="qMRMLNodeComboBox" name="TransformNodeSelector">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLLinearTransformNode</string>
         <string>vtkMRMLBSplineTransformNode</string>
         <string>vtkMRMLGridTransformNode</string>
         <string>vtkMRMLTransformNode</string>
        </stringlist>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="InfoCollapsibleWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Information</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <property name="collapsedHeight">
      <number>14</number>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <widget class="qMRMLTransformInfoWidget" name="TransformInfoWidget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="DisplayEditCollapsibleWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Edit</string>
     </property>
     <property name="contentsLineWidth">
      <number>0</number>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="ctkCollapsibleGroupBox" name="MatrixViewGroupBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Transform Matrix</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>2</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="qMRMLMatrixWidget" name="MatrixWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="decimalsOption">
            <set>ctkDoubleSpinBox::DecimalsByShortcuts</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="qMRMLTransformSliders" name="TranslationSliders">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="qMRMLTransformSliders" name="RotationSliders">
        <property name="Title">
         <string>Rotation</string>
        </property>
        <property name="TypeOfTransform">
         <enum>qMRMLTransformSliders::ROTATION</enum>
        </property>
        <property name="minMaxVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_1">
        <property name="spacing">
         <number>2</number>
        </property>
        <item>
         <widget class="QPushButton" name="IdentityPushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Identity</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="InvertPushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Invert</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="SplitPushButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>Split a composite transform to its components</string>
          </property>
          <property name="text">
           <string>Split</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QGroupBox" name="CoordinateReferenceGroupBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Coordinate Reference</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QRadioButton" name="GlobalRadioButton">
             <property name="text">
              <string>Global</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="LocalRadioButton">
             <property name="text">
              <string>Local</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="DisplayCollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Display</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="qMRMLTransformDisplayNodeWidget" name="TransformDisplayNodeWidget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="TransformedCollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Apply transform</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="TransformableLabel">
          <property name="text">
           <string>Transformable:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qMRMLTreeView" name="TransformableTreeView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>160</width>
            <height>0</height>
           </size>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="headerHidden">
           <bool>true</bool>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLTransformableNode</string>
           </stringlist>
          </property>
          <property name="fitSizeToVisibleIndexes">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QToolButton" name="TransformToolButton">
          <property name="toolTip">
           <string>Apply the active transform to the selected transformable nodes</string>
          </property>
          <property name="text">
           <string>&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="UntransformToolButton">
          <property name="toolTip">
           <string>Remove the active transform from the selected transformed nodes</string>
          </property>
          <property name="text">
           <string>&lt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="HardenToolButton">
          <property name="toolTip">
           <string>Harden transform</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../qSlicerTransformsModule.qrc">
            <normaloff>:/Icons/Transforms.png</normaloff>:/Icons/Transforms.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QLabel" name="TransformedLabel">
          <property name="text">
           <string>Transformed:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qMRMLTreeView" name="TransformedTreeView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>160</width>
            <height>0</height>
           </size>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="headerHidden">
           <bool>true</bool>
          </property>
          <property name="fitSizeToVisibleIndexes">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="BottomSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLMatrixWidget</class>
   <extends>ctkMatrixWidget</extends>
   <header>qMRMLMatrixWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLTransformSliders</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLTransformSliders.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLTreeView</class>
   <extends>QTreeView</extends>
   <header>qMRMLTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLTransformDisplayNodeWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLTransformDisplayNodeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLTransformInfoWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLTransformInfoWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkMatrixWidget</class>
   <extends>QWidget</extends>
   <header>ctkMatrixWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../qSlicerTransformsModule.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TransformableTreeView</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>1206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TransformedTreeView</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>501</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>1206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TransformNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>6</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>TranslationSliders</receiver>
   <slot>setMRMLTransformNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>336</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>RotationSliders</receiver>
   <slot>setMRMLTransformNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>430</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>MatrixWidget</receiver>
   <slot>setMRMLTransformNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TranslationSliders</sender>
   <signal>valuesChanged()</signal>
   <receiver>RotationSliders</receiver>
   <slot>resetUnactiveSliders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>430</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>InvertPushButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>161</x>
     <y>516</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>TransformToolButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>907</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>UntransformToolButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>1014</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>TransformDisplayNodeWidget</receiver>
   <slot>setMRMLTransformNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>10</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>758</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TransformDisplayNodeWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>758</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TransformNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>TransformInfoWidget</receiver>
   <slot>setMRMLTransformNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>10</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TranslationSliders</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTransformsModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TransformInfoWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>611</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
